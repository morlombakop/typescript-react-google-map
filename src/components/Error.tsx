import * as React from 'react'
import styled from 'styled-components'

const Error: React.FC = () => {
  const Container = styled.div`
    margin: 0;
    position: absolute;
    width: 100%;
    top: 50%;
    -ms-transform: translateY(-50%);
    transform: translateY(-50%);
    h1 {
      color: red;
      text-align: center;
    }
    h5 {
      text-align: center;
      font-weight: normal;
    }
  `

  return (
    <Container data-testid="error-container">
      {/* In a normal project these texts will be saved in a separate json file for internationalization */}
      <h1>Error</h1>
      <h5>Please ensure you have an internet connection and reload the page.</h5>
    </Container>
  )
}

export default Error
