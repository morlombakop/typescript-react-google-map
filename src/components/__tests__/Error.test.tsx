import 'jest-dom/extend-expect'
import 'jest-styled-components'
import * as React from 'react'
import { cleanup, render } from '@testing-library/react'
import Error from '../Error'

afterAll(cleanup)

describe('Error component tests', () => {
  const { getByTestId } = render(<Error />)
  const errorContainer = getByTestId('error-container')

  test('Error component renders properly', () => {
    expect(errorContainer).toMatchSnapshot()
    expect(errorContainer).toBeInstanceOf(HTMLDivElement)
    expect(errorContainer.childElementCount).toBe(2)
    expect(errorContainer.firstChild).toBeInstanceOf(HTMLHeadingElement)
    expect(errorContainer.lastChild).toBeInstanceOf(HTMLHeadingElement)
  })

  test('Error component renders with correct style', () => {
    expect(errorContainer).toHaveStyle(`
      margin: 0;
      position: absolute;
      width: 100%;
      top: 50%;
      transform: translateY(-50%);
    `)
    expect(errorContainer.firstChild).toHaveStyle(`
      color: red;
      text-align: center;
    `)
    expect(errorContainer.lastChild).toHaveStyle(`
      text-align: center;
      font-weight: normal;
    `)
  })

  test('Error component renders with correct content', () => {
    expect(errorContainer.firstChild).toHaveTextContent('Error')
    expect(errorContainer.lastChild).toHaveTextContent(
      'Please ensure you have an internet connection and reload the page.'
    )
  })
})
