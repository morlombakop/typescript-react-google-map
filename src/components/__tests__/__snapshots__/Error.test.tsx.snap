// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component tests Error component renders properly 1`] = `
.c0 {
  margin: 0;
  position: absolute;
  width: 100%;
  top: 50%;
  -ms-transform: translateY(-50%);
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 h1 {
  color: red;
  text-align: center;
}

.c0 h5 {
  text-align: center;
  font-weight: normal;
}

<div
  class="c0"
  data-testid="error-container"
>
  <h1>
    Error
  </h1>
  <h5>
    Please ensure you have an internet connection and reload the page.
  </h5>
</div>
`;
